docs/ignoring-files-folders-code.md

```diff
@@ -0,0 +1,186 @@
+---
+
+import MoreHelp from "/src/components/MoreHelp"
+
+
+# Ignoring files, folders, or code
+
+**Ignoring files, folders, or code** means Semgrep will not scan certain defined items within your repository or working directory. Thus no findings are generated.
+
+This is separate from [ignoring a **finding**](../managing-findings), which is a triage operation after a Semgrep scan has found a match.
+
+All Semgrep environments (CLI, CI, and App) adhere to user-defined or Semgrep-defined ignore patterns.
+
+
+## Reference summary
+
+| Method  | Usage    | Examples |
+|:--------|:---------|:---------|
+| To ignore blocks of code: `nosemgrep` | Create a comment, followed by a space (` `), followed by `nosemgrep` at the first line or preceding line of the pattern match. | ` // nosemgrep` &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; `// nosemgrep: rule-id` <br /> `# nosemgrep` |
+| To ignore files and folders: `.semgrepignore` | Create a `.semgrepignore` file and add patterns for files and folders there. Patterns follow `.gitignore` syntax with some caveats. See [Defining files and folders in `.semgrepignore`](#defining-files-and-folders-in-semgrepignore). | [`.semgrepignore` sample file](https://raw.githubusercontent.com/returntocorp/semgrep/develop/semgrep/semgrep/templates/.semgrepignore) |
+
+
+## Understanding Semgrep defaults
+
+Without user customization, Semgrep refers to the following to define ignored files and folders:
+
+* Semgrep's default `.semgrepignore` file
+* Your repository's `.gitignore` file (if it exists)
+
+The following is **Semgrep's default `.semgrepignore` file**:
+
+```
+DEFAULT_SEMGREPIGNORE_TEXT
+```
+
+:::caution
+
+Note that the default `.semgrepignore` file is opinionated and causes Semgrep to skip these folders:
+
+* `/tests`, `/test`
+* `/vendors`
+
+:::
+
+
+## Files, folders and code beyond Semgrep's scope
+
+Semgrep will ignore files beyond the scope of Semgrep's use.
+
+* Large files (maximum file size defaults to 1MB)
+* Binary files
+* Unknown file extensions (file extensions not matched with any supported programming language)
+
+Large files and unknown file extensions are included or excluded through command line flags (See [CLI reference](https://semgrep.dev/docs/cli-usage/)). Binary files are never scanned.
+
+This document defines **files, folders and code** as those that are **relevant to a Semgrep scan**. For example, `.jpg` files are not a part of Semgrep's scope and therefore are not part of the scope of this document.
+
+
+## Customizing ignore behavior
+
+Semgrep provides several methods to customize ignore behavior. Refer to the following table to see which method suits your goal:
+
+| Goal | Method |
+|:---- |:------ |
+| To scan all files each time you run Semgrep (no files will be ignored) | Create an empty `.semgrepignore` file. An empty `.semgrepignore` will result in patterns within `.gitignore` **to be scanned**. |
```

Comment:
Where should this .semgrepignore file be placed?
(maybe it’s better to state its required location elsewhere, and not in this table)

Reply:
I mention it twice in my update (both tables) so people don't miss it :) Thanks for the catch!